core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -134,6 +134,10 @@ public boolean handle(int code) {
             // we've already logged this.
         }
 
+        if (Constants.LINUX) {
+            JNANatives.maxNumberOfProcesses();
```

Comment:
we should rename all the code in this PR to numberOfThreads and say nothing about processes.

I think its really important:
* we don't care about max number of `processes` and in fact drop it to 0 for all operating systems intentionally.
* we do care about max number of `threads`, which is RLIMIT_NPROC on linux. But this is super confusing because RLIMIT_NPROC has a different meaning everywhere else on the planet (where it means processes).